docs/releases/upgrading/upgrade-v12.0.md

```diff
@@ -0,0 +1,215 @@
+    using the existing infrastructure to v11 and afterwards continue from here.
+
+## Upgrade Steps
+
+!!! warning "Upgrade your invenio-cli"
+
+    Make sure you have the latest `invenio-cli` installed. For InvenioRDM v12 it
+    is at least v1.2.0
+
+    ```bash
+    $ invenio-cli --version
+    invenio-cli, version 1.2.0
+    ```
+
+!!! info "Virtual environments"
+
+    In case you are not inside a virtual environment, make sure that you prefix each `invenio`
+    command with `pipenv run`.
+
+
+**Local development**
+
+Changing the Python version in your development environment highly
+depends on your setup, and there is no golden rule.
+One example would be to use [PyEnv](https://github.com/pyenv/pyenv).
+
+You should delete your virtualenv before running `invenio-cli` or `pipenv` commands below.
+
+!!! warning "Risk of losing data"
+
+    Your virtual env folder contains uploaded files in InvenioRDM, in `var/instance/data`.
+    If you need to keep such files, make sure you copy them over to the new virtual env in the same location.
+
+### Upgrade InvenioRDM
+
+Make sure that your virtual env is now running with at least Python 3.9. (InvenioRDM v12
+is also tested with Python 3.12.)
+
+There are two options to upgrade your system.
+
+#### Option 1
+
+This approach upgrades the dependencies in place. Your virtual environment for the
+v11 version will be gone afterwards.
+
+```bash
+cd <my-site>
+
+# Upgrade to InvenioRDM v12
+invenio-cli packages update 12.0.0
+pipenv uninstall flask-babelex
+
+# Re-build assets
+invenio-cli assets build
+```
+
+#### Option 2
+
+This approach will create a new virtual environment and leaves the v11 as it is.
+If you are using a docker image on your production instance this will be the
+option you chose.
+
+##### Step 1
+- create a new virtual environment
+- activate your new virtual environment
+- install `invenio-cli` by `pip install invenio-cli`
+
+##### Step 2
+Update the file `<my-site>/Pipfile`.
+
+```diff
+[packages]
+---invenio-app-rdm = {extras = [...], version = "~=11.0.0"}
++++invenio-app-rdm = {extras = [...], version = "~=12.0.0"}
+```
+
+Due to a dependency upgrade, update the following line.
+```diff
+---my-site = {editable="True", path="./site"}
++++my-site = {editable=true, path="./site"}
+```
+
+##### Step 3
+Update the `Pipfile.lock` file
+
+```bash
+invenio-cli packages lock
+```
+
+##### Step 4
+install InvenioRDM v12
+
+```bash
+invenio-cli install
+```
+
+
+### Database migration
+
+Execute the database migration:
+
+```bash
+# Execute the database migration
+invenio alembic upgrade
+```
+
+### Declare usage statistics processing queues
+
+```shell
+invenio queues declare
+```
+
+### Update indices mappings
+
+```shell
+invenio index update communities-communities-v1.0.0
+invenio index update rdmrecords-drafts-draft-v5.0.0
+invenio index update rdmrecords-records-record-v5.0.0
+```
+
+CAVEAT: this is not working because of a permission problem in
```

Comment:
So should we just provide instructions to destroy the index? I don't see why we'd provide commands that don't work.

Reply:
the instructions are some lines below the question is if we should remove that part completely. i will remove the whole `update indices` section including the text. it is not necessary for the migration to explain that the `index index destroy` keeps the statistics because at this point no statistics exists but i will move that to the section `reindex` so that the information is there. in my opinion it is a important information which could be overlooked if it is written somewhere else and here it will be seen because the upgrade steps have to be gone through step by step.